openmc/filter_expansion.py

```diff
@@ -463,3 +463,136 @@ def to_xml_element(self):
         subelement.text = str(self.r)
 
         return element
+
+class ZernikeRadialFilter(ExpansionFilter):
+    r"""Score radial Zernike expansion moments in space up to specified order.
+
+    The Zernike polynomials are defined the same as in ZernikeFilter.
+
+    .. math::
+        Z_n^m(\rho, \theta) = R_n^m(\rho) \cos (m\theta), \quad m > 0
+
+        Z_n^{m}(\rho, \theta) = R_n^{m}(\rho) \sin (m\theta), \quad m < 0
+
+        Z_n^{m}(\rho, \theta) = R_n^{m}(\rho), \quad m = 0
+
+    where the radial polynomials are
+
+    .. math::
+        R_n^m(\rho) = \sum\limits_{k=0}^{(n-m)/2} \frac{(-1)^k (n-k)!}{k! (
+        \frac{n+m}{2} - k)! (\frac{n-m}{2} - k)!} \rho^{n-2k}.
+
+    With this definition, the integral of :math:`(Z_n^m)^2` over the unit disk
+    is :math:`\frac{\epsilon_m\pi}{2n+2}` for each polynomial where :math:`\epsilon_m` is
+    2 if :math:`m` equals 0 and 1 otherwise.
+
+    If there is only radial dependency, the polynomials are integrated over the azimuthal angles. The only terms left are :math:`Z_n^{0}(\rho, \theta) = R_n^{0}(\rho)`. Note that :math:`n` could only be even orders. Therefore, for a radial Zernike polynomials up to order of :math:`n`, there are :math:`\frac{n}{2} + 1` terms in total.
+
+    Parameters
+    ----------
+    order : int
+        Maximum radial Zernike polynomial order
+    x : float
+        x-coordinate of center of circle for normalization
+    y : float
+        y-coordinate of center of circle for normalization
+    r : int or None
+        Radius of circle for normalization
+
+    Attributes
+    ----------
+    order : int
+        Maximum radial Zernike polynomial order
+    x : float
+        x-coordinate of center of circle for normalization
+    y : float
+        y-coordinate of center of circle for normalization
+    r : int or None
+        Radius of circle for normalization
+    id : int
+        Unique identifier for the filter
+    num_bins : int
+        The number of filter bins
+
+    """
+
+    def __init__(self, order, x=0.0, y=0.0, r=1.0, filter_id=None):
+        super().__init__(order, filter_id)
+        self.x = x
+        self.y = y
+        self.r = r
```

Comment:
Most of the methods/properties in this class are exactly the same as for `ZernikeFilter`. It would be cleaner to just subclass ZernikeFilter and then only modify the methods that need to be changed. In fact, I think the only thing that would need to be changed is the setter for `order`. Everything else can probably be omitted.